web/crux-ui/src/components/deployments/add-deployment-card.tsx

```diff
@@ -0,0 +1,244 @@
+import DyoButton from '@app/elements/dyo-button'
+import { DyoCard } from '@app/elements/dyo-card'
+import DyoChips from '@app/elements/dyo-chips'
+import { DyoHeading } from '@app/elements/dyo-heading'
+import { DyoInput } from '@app/elements/dyo-input'
+import { DyoLabel } from '@app/elements/dyo-label'
+import DyoMessage from '@app/elements/dyo-message'
+import DyoTextArea from '@app/elements/dyo-text-area'
+import { apiErrorHandler, defaultTranslator } from '@app/errors'
+import useDyoFormik from '@app/hooks/use-dyo-formik'
+import useTeamRoutes from '@app/hooks/use-team-routes'
+import {
+  CreateDeployment,
+  Deployment,
+  DyoApiError,
+  DyoNode,
+  Project,
+  Version,
+  projectNameToDeploymentPrefix,
+} from '@app/models'
+import { fetcher, sendForm } from '@app/utils'
+import { createFullDeploymentSchema } from '@app/validations'
+import useTranslation from 'next-translate/useTranslation'
+import { useEffect } from 'react'
+import useSWR from 'swr'
+
+interface AddDeploymentCardProps {
+  className?: string
+  onAdd: (deploymentId: string) => void
+  onDiscard: VoidFunction
+}
+
+const AddDeploymentCard = (props: AddDeploymentCardProps) => {
+  const { className, onAdd, onDiscard } = props
+
+  const { t } = useTranslation('deployments')
+  const routes = useTeamRoutes()
+
+  const { data: nodes, error: fetchNodesError } = useSWR<DyoNode[]>(routes.node.api.list(), fetcher)
+  const { data: projects, error: fetchProjectsError } = useSWR<Project[]>(routes.project.api.list(), fetcher)
+
+  const handleApiError = apiErrorHandler((stringId: string, status: number, dto: DyoApiError) => {
+    if (dto.error === 'rollingVersionDeployment' || dto.error === 'alreadyHavePreparing') {
+      onAdd(dto.value)
+
+      return {
+        toast: dto.description,
+        toastOptions: {
+          className: '!bg-warning-orange',
+          duration: 5000,
```

Comment:
`5000` looks like a magic number, for better readability, we try to outsource "random" numbers and use contacts. Could you do the same with this?